pkg/vault/test/authorizer/TimelockAuthorizerExecute.test.ts

```diff
@@ -0,0 +1,431 @@
+
+      const scheduledExecution = await authorizer.getScheduledExecution(id);
+      expect(scheduledExecution.executed).to.be.true;
+
+      expectEvent.inIndirectReceipt(await receipt.wait(), authenticatedContract.interface, 'ProtectedFunctionCalled', {
+        data: functionData,
+      });
+    });
+
+    it('emits an event', async () => {
+      const id = await schedule();
+      await advanceTime(delay);
+      const receipt = await authorizer.execute(id, { from: executor });
+
+      expectEvent.inReceipt(await receipt.wait(), 'ExecutionExecuted', {
+        scheduledExecutionId: id,
+      });
+    });
+
+    it('cannot be executed twice', async () => {
+      const id = await schedule();
+      await advanceTime(delay);
+      await authorizer.execute(id, { from: executor });
+      await expect(authorizer.execute(id, { from: executor })).to.be.revertedWith('ACTION_ALREADY_EXECUTED');
+    });
+
+    it('reverts if action was cancelled', async () => {
+      const id = await schedule();
+      await advanceTime(delay);
+      await authorizer.cancel(id, { from: user });
+      await expect(authorizer.execute(id, { from: executor })).to.be.revertedWith('ACTION_ALREADY_CANCELLED');
+    });
+
+    it('reverts if the delay has not passed', async () => {
+      const id = await schedule();
+      await expect(authorizer.execute(id, { from: executor })).to.be.revertedWith('ACTION_NOT_YET_EXECUTABLE');
+    });
+
+    it('reverts if the scheduled id is invalid', async () => {
+      await expect(authorizer.execute(100)).to.be.revertedWith('ACTION_DOES_NOT_EXIST');
+    });
+  });
+
+  describe('cancel', () => {
+    const delay = DAY;
+
+    sharedBeforeEach('grant protected function permission with delay', async () => {
+      // We must set a delay for the `setAuthorizer` function as well to be able to give one to `protectedFunction`
+      const setAuthorizerAction = await actionId(vault, 'setAuthorizer');
+      await authorizer.scheduleAndExecuteDelayChange(setAuthorizerAction, delay, { from: root });
+
+      const protectedFunctionAction = await actionId(authenticatedContract, 'protectedFunction');
+      await authorizer.scheduleAndExecuteDelayChange(protectedFunctionAction, delay, { from: root });
+      await authorizer.grantPermission(protectedFunctionAction, user, authenticatedContract, { from: root });
+    });
+
+    const schedule = async (): Promise<number> => {
+      const data = authenticatedContract.interface.encodeFunctionData('protectedFunction', ['0x']);
+      const id = await authorizer.schedule(authenticatedContract, data, [], { from: user });
+      await authorizer.addCanceler(id, canceler, { from: root });
+      return id;
+    };
+
+    it('cancel the action', async () => {
+      const id = await schedule();
+      await authorizer.cancel(id, { from: canceler });
+
+      const scheduledExecution = await authorizer.getScheduledExecution(id);
+      expect(scheduledExecution.cancelled).to.be.true;
+    });
```

Comment:
I'd do this with both a specific and a global canceler, since we otherwise never have a global canceler use its permission (we only check that `isCanceler` returns true).  We could also add a third test with `root`, which will be able to do it even if it never received the permission.